app/packs/src/apps/mydb/elements/details/samples/SampleDetails.js

```diff
@@ -739,59 +748,121 @@ export default class SampleDetails extends React.Component {
     )
   }
 
-  addManualCas(e) {
-    DetailActions.updateMoleculeCas(this.props.sample, e.value);
+  // addManualCas(e) {
+  //   DetailActions.updateMoleculeCas(this.props.sample, e.value);
+  // }
+
+  addHypenToCas(cas, boolean) {
+    const { sample } = this.state;
+    const filterCas = cas.replace(/-/g, '');
+    const firstHypenIndex = filterCas.length - 3;
+    const secondHypenIndex = filterCas.length - 1;
+    console.log(filterCas);
+    const firstPart = filterCas.substring(0, firstHypenIndex);
+    const secondPart = filterCas.substring(firstHypenIndex, secondHypenIndex);
+    const thirdPart = filterCas.substring(secondHypenIndex, filterCas.length);
+    const format = `${firstPart}-${secondPart}-${thirdPart}`;
+    if (boolean) {
+      sample.xref = { ...sample.xref, cas: format };
+      this.setState({ sample });
+    }
+    console.log(format);
+    const checkFormat = format.match(/([0-9]{2,7})-([0-9]{2})-[0-9]/);
+    console.log(checkFormat);
+    return checkFormat;
+  }
+
+  checkCasValidation(cas, boolean) {
```

Comment:
As far as I can tell, this  method does (at least) two things:

1. figure out if the string that's passed in for the `cas` parameter is a CAS or a Smiles string.
2. if it's a CAS, validate it

Would it be an idea to split this method into two, according to these two tasks?
E.g., `casOrSmiles` and `validateCas`?